testcases/open_posix_testsuite/conformance/interfaces/aio_cancel/5-1.c

```diff
@@ -137,8 +139,35 @@ int main(void)
 		}
 	} while (in_progress);
 
-	if (!check_one)
-		return PTS_UNRESOLVED;
+	if (!check_one_done) {
+		return NO_FINISHED;
+	} else if (!check_one_canceled) {
+		return NO_CANCELED;
+	}
+
```

Comment:
Why do we have to return different statuses when we don't really make use of that information?

Can we rather return 0 on success and 1 when we have to retry? That would simplify the code here as well as in the main loop.

Reply:
I think that is a relict of my debugging code. I think it would still be useful to distinguish between the two cases and in case of a failure print a message with the cause of the failure, for debugging. This part is missing from the main function at the moment, but would require the different return codes.

Reply:
I added an output with the retry reason. This should help when debugging a failing test.